util/preproc.q

```diff
@@ -166,19 +168,19 @@ oneHot.fit:{[tab;symCols]
   if[(::)~symCols;symCols:i.findCols[tab;"s"]];
   mapVals:asc each distinct each tab symCols,:(); 
   mapDict:symCols!mapVals;
-  predict:oneHot.predict[;;mapDict];
+  predict:oneHot.predict mapDict;
   `modelInfo`predict!(mapDict;predict)
   }
 
 // @kind function
 // @category preprocessing
 // @fileoverview Encode categorical features using one-hot encoded fitted model
+// @return {tab} One-hot encoded representation of categorical data
+// @params mapDict {dict} Map cateogorical values to their encoded values
 // @param tab {tab} Numerical and non numerical data
 // @param symDict {dict} Keys indicate the columns in the table to be encoded,
 //   values indicate what mapping to use when encoding 
-// @params mapDict {dict} Map cateogorical values to their encoded values
-// @return {tab} One-hot encoded representation of categorical data
-oneHot.predict:{[tab;symDict;mapDict]
+oneHot.predict:{[mapDict;tab;symDict]
```

Comment:
Once this has been changed to take the output from the `oneHot.fit` function the naming of `mapDict` should probably change to `modelInfo`/`config` in line with similar function.

The same will hold for the lexigraphical encoding below.

Reply:
And also the `labelEncode.[fit/predict/applyLabelEncode]` functionality too

Reply:
And also the `labelEncode.[fit/predict/applyLabelEncode]` functionality too